### PR TITLE
Add <SearchableList /> component

### PR DESCRIPTION
**Pull request by kntsoriano**
## Description
This PR adds the <SearchableList /> component which allows us to have lists that can be paginated, filtered, sorted, and searched. In this PR, I've used this component to add the <SolutionsList /> component in the journey page.

Once the API endpoints for the several lists in the journey page is done, I will make an end to end integration. For the meanwhile, I'm opening this PR to get feedback for my implementation.

### CODE REVIEW

File: app/javascript/components/journey/SolutionFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { usePanel } from '../../hooks/use-panel'
+import { GraphicalIcon } from '../common'
+
+type SolutionFilterOption = {
+  value: string
+  title: string
+}
+
+const OPTIONS: SolutionFilterOption[] = [
```
Comment (kntsoriano) [inline app/javascript/components/journey/SolutionFilter.tsx:10]: I dont know what the real filter options are.

File: app/javascript/components/journey/SolutionResults.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  sort: string
+}): JSX.Element => {
+  return (
+    <div>
+      <div className="results-title-bar">
+        <h3>
+          Showing {results.length} {pluralize('solutions', results.length)}
+        </h3>
+        <div className="c-select order">
+          <select onChange={(e) => setSort(e.target.value)} value={sort}>
```
Comment (kntsoriano) [inline app/javascript/components/journey/SolutionResults.tsx:21]: I don't know what the real sort options are.

File: app/javascript/components/journey/SolutionsList.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { SolutionResults } from './SolutionResults'
+import { SearchableList } from '../common/SearchableList'
+
+export const SolutionsList = ({
+  endpoint,
+}: {
+  endpoint: string
+}): JSX.Element => {
+  return (
+    <SearchableList
```
Comment (kntsoriano) [inline app/javascript/components/journey/SolutionsList.tsx:12]: This is the generic component used in a specific context.

File: app/javascript/components/journey/SolutionFilter.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    value: 'functional',
+    title: 'Functional',
+  },
+]
+
+export const SolutionFilter = ({
+  value,
+  setFilter,
+}: {
+  value: string[]
```
Comment (ErikSchierboom) [inline app/javascript/components/journey/SolutionFilter.tsx]: Should this be plural?

```suggestion
  values: string[]
```

File: app/javascript/components/journey/SolutionResults.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+}: {
+  results: SolutionProps[]
+  setSort: (sort: string) => void
+  sort: string
+}): JSX.Element => {
+  return (
+    <div>
+      <div className="results-title-bar">
+        <h3>
+          Showing {results.length} {pluralize('solutions', results.length)}
```
Comment (ErikSchierboom) [inline app/javascript/components/journey/SolutionResults.tsx]: Should this be:

```suggestion
          Showing {results.length} {pluralize('solution', results.length)}
```

File: test/javascript/components/journey/SolutionsList.test.js [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+  render(<SolutionsList endpoint="https://exercism.test/solutions" />)
+
+  expect(await screen.findByText('Showing 2 solutions')).toBeInTheDocument()
+  expect(await screen.findByText('Lasagna')).toBeInTheDocument()
+  expect(await screen.findByText('Bob')).toBeInTheDocument()
+
+  server.close()
+})
+test('paginates solutions', async () => {
```
Comment (ErikSchierboom) [inline test/javascript/components/journey/SolutionsList.test.js:46]: A little visual separation makes it easier to identify the different tests I feel

```suggestion

test('paginates solutions', async () => {
```

File: test/javascript/components/journey/SolutionsList.test.js [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+  render(<SolutionsList endpoint="https://exercism.test/solutions" />)
+
+  userEvent.click(await screen.findByRole('button', { name: 'Go to page 2' }))
+
+  expect(await screen.findByText('Bob')).toBeInTheDocument()
+
+  server.close()
+})
+test('searches solutions', async () => {
```
Comment (ErikSchierboom) [inline test/javascript/components/journey/SolutionsList.test.js:83]:
```suggestion

test('searches solutions', async () => {
```

File: test/javascript/components/journey/SolutionsList.test.js [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+
+  render(<SolutionsList endpoint="https://exercism.test/solutions" />)
+
+  userEvent.type(screen.getByPlaceholderText('Search for an exercise'), 'Bob')
+
+  expect(await screen.findByText('Bob')).toBeInTheDocument()
+
+  server.close()
+})
+test('filters solutions', async () => {
```
Comment (ErikSchierboom) [inline test/javascript/components/journey/SolutionsList.test.js:124]:
```suggestion

test('filters solutions', async () => {
```

**Review by ErikSchierboom — APPROVED** [review]

File: app/javascript/components/journey/SolutionFilter.tsx (hunk 1 above)
Comment (iHiD) [reply app/javascript/components/journey/SolutionFilter.tsx:10]: Me neither!!

**Review by iHiD — APPROVED** [review]
Sweet :)

**Comment by kntsoriano** [comment]
@ErikSchierboom Thanks :) I've applied your requested changes.

**Pull request merged**